src/theory/strings/theory_strings.cpp

```diff
@@ -3809,6 +3842,56 @@ void TheoryStrings::registerTerm( Node n, int effort ) {
   }
 }
 
+void TheoryStrings::sendInternalInference(std::vector< Node >& exp, Node conc, const char * c)
+{
+  if( conc.getKind()==AND )
+  {
+    for( const Node& cc : conc )
+    {
+      sendInternalInference( exp, cc, c );
+    }
+    return;
+  }
+  Node lit = conc.getKind()==NOT ? conc[0] : conc;
+  bool pol = conc.getKind()!=NOT;
```

Comment:
Suggestion: It would be slightly more elegant to first determine `pol` and then use that in the condition for `lit`.